### PR TITLE
Update to CosmWasm v1.0.0 and add arch64 support to Dockerfile

### PR DESCRIPTION
**Pull request by krebernisak**

### CODE REVIEW

**Review by anirudhwarrier — APPROVED** [review]
Working on M1

File: go.mod [hunk 1]
```diff
@@ -90,6 +90,8 @@ require (
 	gopkg.in/guregu/null.v4 v4.0.0
 )
 
+require github.com/cosmos/gorocksdb v1.2.0 // indirect
```
Comment (jmank88) [inline go.mod:93]: ^ This one escaped the block with the rest.

File: go.mod [hunk 2]
```diff
@@ -349,7 +350,10 @@ replace github.com/dgrijalva/jwt-go => github.com/form3tech-oss/jwt-go v3.2.1+in
 replace github.com/gogo/protobuf => github.com/regen-network/protobuf v1.3.3-alpha.regen.1
 
 // needed to address mismatch between cosmosSDK and hdevalence/ed25519consensus
-replace filippo.io/edwards25519 => filippo.io/edwards25519 v1.0.0-beta.3
+replace filippo.io/edwards25519 => filippo.io/edwards25519 v1.0.0-rc.1
+
+// updating CosmWasm to v1.0.0 which brings ARM support
+replace github.com/CosmWasm/wasmvm => github.com/CosmWasm/wasmvm v1.0.0
```
Comment (jmank88) [inline go.mod:356]: Does `integration-tests/go.mod` need the same/similar?

File: README.md [hunk 3]
```diff
@@ -50,23 +50,14 @@ For the latest information on setting up a development environment, see the [Dev
 
 ### Mac M1/ARM64 [EXPERIMENTAL]
```
Comment (samsondav) [inline README.md:51]: I'd say we are good to remove the "experimental" tag from this. We have multiple developers running it compiled natively for ARM for months without issue.

**Review by samsondav — CHANGES_REQUESTED** [review]

File: go.mod (hunk 2 above)
Comment (krebernisak) [reply go.mod:356]: I don't think so. 

This is an indirect dependency of `github.com/smartcontractkit/chainlink-terra`, which isn't present in local `integration-tests`.

Comment (krebernisak) [reply go.mod:354]: Yeah, it looks it does. We need to untangle that dependency tree.

**Review by jmank88 — DISMISSED** [review]

**Review by vyzaldysanchez — DISMISSED** [review]

**Review by HenryNguyen5 — DISMISSED** [review]

**Review by reductionista — APPROVED** [review]
Tested on M1 platform, works!  LGTM assuming it still works in CI and other platforms.

**Review by jmank88 — DISMISSED** [review]

**Review by samsondav — APPROVED** [review]

**Review by jmank88 — APPROVED** [review]

**Review by vyzaldysanchez — APPROVED** [review]

**Pull request merged**